### PR TITLE
chore: upgraded to Gradle 7

### PR DESCRIPTION
**Pull request by quintesse**

### CODE REVIEW

**Comment by quintesse** [comment]
In the end I used that plugin you found that can generate the `BuildConfig` class. It was the simplest change.

**Comment by maxandersen** [comment]
add line to ignore generated sources and now merging.

File: build.gradle [hunk 1]
```diff
@@ -94,6 +94,7 @@ spotless {
 		importOrder 'java', 'javax', 'org', 'com', 'dev.jbang', ''
 		removeUnusedImports()
 		eclipse().configFile "misc/eclipse_formatting_nowrap.xml"
```
Comment (quintesse) [inline build.gradle:96]: Aaah, I had put my exclude in the wrong section. Duh.

Comment (maxandersen) [reply build.gradle:96]: Yeah. I was getting upset why it wasn't working ...until i noticed there was multiple sections :)

**Pull request merged**